### PR TITLE
Fix uninitialized `fused_shift_log_probas` in `forward()`

### PR DESCRIPTION
**Pull request by aaronkollasch**
When using [`model.generate()`](https://huggingface.co/docs/transformers/v4.24.0/en/main_classes/text_generation#transformers.generation_utils.GenerationMixin.generate), we were getting an error that `fused_shift_log_probas` was uninitialized when `labels` is `None`.

This PR ensures that `fused_shift_log_probas` is initialized as `None` regardless of the value of `labels`.

@hanspinner

### CODE REVIEW

**Comment by pascalnotin** [comment]
Thanks for the PR @aaronkollasch. The fix you are suggesting works but just wanted to flag that when using `model.generate` you are likely not factoring in the MSA retrieval and instead generate fully with the autoregressive transformer. For instance, I don't think `model.generate` is using `fused_shift_log_probas` (which is a field I had introduced to aggregate the autoregressive predictions with the ones from the retrieved MSA / is not part of the standard HuggingFace Transformers codebase). Perhaps we should extend `model.generate` or modify the `model.forward` method so that -- if we do want to use retrieval at generation time -- then `lm_logits` is overwritten (I have not looked at model.generate but I suspect it uses  `lm_logits` to generate sequences).

**Comment by aaronkollasch** [comment]
Thanks @pascalnotin,
Yeah we were intending to generate just with the autoregressive transformer portion of the model, at least for now.

It looks like the easiest place to combine the logits is in the output of `model.forward` but I'm not 100% sure.
After the initial seed sequence (where `lm_logits` has shape `N x L_seed x C`), it produces `lm_logits` of  shape`N x 1 x C` for each subsequent sampling step/position, so the forward method would have to keep track of the position that is being generated (or it may be inferred from `past_key_values`). This is not exactly trivial as the autoregressive model can generate sequences with insertions and deletions.

**Comment by pascalnotin** [comment]
Good point re: indels, although there are ways to adapt the generation code to account for that (not currently handled by model.generate). Approving the PR to support generation with the autoregressive transformer only for now.
